### PR TITLE
Fix missing includes with recent compiler versions

### PR DESCRIPTION
**Pull request by maxnoe**
Trying to compile this using recent compilers I ran into some issues of missing header includes, also in the docopt code, which was fixed by updating to their latest release.

### CODE REVIEW

**Comment by relleums** [comment]
Thank you Max! I will have a look.

**Comment by maxnoe** [comment]
Note that the github actions config I put in place does not yet cover everything the old travis ci config did

File: merlict/HomTra3.cpp
Comment (relleums) [inline merlict/HomTra3.cpp:1]: Good thing that operator= is not needed.

File: merlict/scenery/ResourceMap.h
Comment (relleums) [inline merlict/scenery/ResourceMap.h:1]: std::out_of_range is defined in <stdexcept>. Got it

File: merlict/small_ball.h
Comment (relleums) [inline merlict/small_ball.h:1]: uint64_t is in cstdint.

File: merlict/tests/FresnelRefractionAndReflectionTest.cpp
Comment (relleums) [inline merlict/tests/FresnelRefractionAndReflectionTest.cpp:1]: I wonder what is the cause here. On my machine it still passes with 'Approx(outgoing_to_normal).margin(1e-323)'. That is the float64 limit. Anyhow. Approx is fine.

File: merlict/tests/catch.hpp
Comment (relleums) [inline merlict/tests/catch.hpp:1]: version update

File: merlict_corsika/eventio.h
Comment (relleums) [inline merlict_corsika/eventio.h:1]: ok

File: merlict_viewer/FlyingCamera.cpp
Comment (relleums) [inline merlict_viewer/FlyingCamera.cpp:1]: Looks like I was calling into a c-library. The internet confirms that this 'API 1.0' is no longer shipped. So good thing this gets updated.

File: merlict_viewer/FlyingCamera.h
Comment (relleums) [inline merlict_viewer/FlyingCamera.h:1]: Adopt modern c++-ish opencv API

File: docopt/.gitignore
Comment (relleums) [inline docopt/.gitignore:1]: Update docopt version. Good thing to do.

File: docopt/.travis.yml
Comment (relleums) [inline docopt/.travis.yml:1]: update docopt

File: docopt/CMakeLists.txt
Comment (relleums) [inline docopt/CMakeLists.txt:1]: update docopt

File: docopt/README.rst
Comment (relleums) [inline docopt/README.rst:1]: update docopt

File: docopt/docopt.cpp
Comment (relleums) [inline docopt/docopt.cpp:1]: update docopt

File: docopt/docopt.h
Comment (relleums) [inline docopt/docopt.h:1]: update docopt

File: docopt/docopt.pc.in
Comment (relleums) [inline docopt/docopt.pc.in:1]: update docopt.

File: docopt/docopt_private.h
Comment (relleums) [inline docopt/docopt_private.h:1]: update docopt

File: docopt/docopt_util.h
Comment (relleums) [inline docopt/docopt_util.h:1]: update docopt

File: docopt/docopt_value.h
Comment (relleums) [inline docopt/docopt_value.h:1]: update docopt

File: .github/workflows/ci.yml
Comment (relleums) [inline .github/workflows/ci.yml:1]: I am not used to this generic ci.yml yet. So I better get used to it now. Thanks for updating.

**Review by relleums — APPROVED** [review]
This effort boosts the 'merlict_development_kit '. Updating dependencies and fixing includes is very welcome. Thank you!

File: merlict/HomTra3.cpp
Comment (maxnoe) [reply merlict/HomTra3.cpp:1]: It is needed, but the default implementation works!

File: .github/workflows/ci.yml
Comment (maxnoe) [reply .github/workflows/ci.yml:1]: It's not generic, its for Github Actions only

**Pull request merged**